packages/flutter/lib/src/rendering/custom_layout.dart

```diff
@@ -206,14 +207,13 @@ abstract class MultiChildLayoutDelegate {
     return DiagnosticsProperty<RenderBox>('${childParentData.id}', child);
   }
 
-  void _callPerformLayout(Size size, RenderBox firstChild) {
+  void _callPerformLayout(Size size, RenderBox? firstChild) {
     // A particular layout delegate could be called reentrantly, e.g. if it used
     // by both a parent and a child. So, we must restore the _idToChild map when
     // we return.
-    final Map<Object, RenderBox> previousIdToChild = _idToChild;
+    final Map<Object, RenderBox>? previousIdToChild = _idToChild;
 
-    // TODO(ianh): make the next line final
-    /*late*/ Set<RenderBox>/*!*/ debugPreviousChildrenNeedingLayout;
+    Set<RenderBox>? debugPreviousChildrenNeedingLayout;
```

Comment:
this seems like something we should be able to make late final non-nullable

Reply:
Not really because the first call. This would need a refactoring and to introduce a bool to know if it has already been called.

Reply:
k